second/homework_018.py

```
@@ -0,0 +1,183 @@
+
+    def id(self):
+        return self._id
+
+    def name(self):
+        return self._name
+
+    def company(self):
+        return self._company
+
+    def workers(self):
+        return self._workers
+
+    def add_worker(self, worker):
+        if isinstance(worker, Worker):
+            self._workers.append(worker)
+        else:
+            raise ValueError("Invalid worker instance")
+
+    def remove_worker(self, worker):
+        if worker in self._workers:
+            self._workers.remove(worker)
+
+    def __str__(self):
+        return f"Boss(id={self._id}, name={self._name}, company={self._company})"
+
+class Worker:
+    boss = WorkerBossDescriptor()
+
+    def __init__(self, id_: int, name: str, company: str, boss=None):
+        self._id = id_
+        self._name = name
+        self._company = company
+        self._boss = None
+        # Use the descriptor for boss to ensure proper validation
+        self.boss = boss
+
+    def id(self):
+        return self._id
+
+    def name(self):
+        return self._name
+
+    def company(self):
+        return self.company
+
+    def __str__(self):
+        return f"Worker(id={self._id}, name={self._name}, company={self._company}"
+
+
+boss1 = Boss(1, "Andriy Verevsky", "Kernel")
+boss2 = Boss(2, "Andreas Flodström", "Beetroot academy")
+
+worker1 = Worker(101, "Maria Dutton", "Kernel", boss1)
+worker2 = Worker(102, "Steven Reynolds", "Beetroot academy")
+
+print(boss1.workers)  # Output: [<__main__.Worker object at ...>]
+print(boss2.workers)  # Output: [<__main__.Worker object at ...>]
+
+print(boss1.id())
+print(boss2.id())
+print(worker1)
+print(worker2)
+print(boss1)
+print(boss2)
+
+
+# Task 3
+from functools import wraps
+class TypeDecorators:
+    @staticmethod
+    def to_int(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            result = func(*args, **kwargs)
+            try:
+                return int(result)
+            except (ValueError, TypeError):
```

No, you should not ignore the errors. I you encounter ValueError or TypeError - they should be propagated (i.e., you should not catch and ignore them). If you ignore errors - your programs become unpredictable and the problem would arise later in the execution flow anyway